From ddbdfffada525cef7852d72395a8bfc092c3d624 Mon Sep 17 00:00:00 2001
From: lijuan <juan.li@spacemit.com>
Date: Sat, 22 Jun 2024 10:44:47 +0800
Subject: [PATCH 0763/1200] kconfig: add config ARCH_FORCE_MAX_ORDER to fix
 default max_order

1. fix default max_order 11 to avoid kmalloc failed when enable kasan
inline,eg: 8852be.ko,8852bs.ko

err log:
insmod: can't insert '/lib/modules/6.6.31+/kernel/drivers/net/wireless/realtek/rtl8852be/8852be.ko': Cannot allocate memory

Signed-off-by: lijuan <juan.li@spacemit.com>
Change-Id: If786a566b4ed14415a98614b9eb97c773a6d7e3c
---
 arch/riscv/Kconfig | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/riscv/Kconfig b/arch/riscv/Kconfig
index 4ed238261f2a..f0e3ae64e0e7 100644
--- a/arch/riscv/Kconfig
+++ b/arch/riscv/Kconfig
@@ -300,6 +300,19 @@ config KASAN_SHADOW_OFFSET
 	default 0xdfffffff00000000 if 64BIT
 	default 0xffffffff if 32BIT
 
+config ARCH_FORCE_MAX_ORDER
+	int "Order of maximal physically contiguous allocations"
+	default "11"
+	help
+	  The kernel page allocator limits the size of maximal physically
+	  contiguous allocations. The limit is called MAX_ORDER and it
+	  defines the maximal power of two of number of pages that can be
+	  allocated as a single contiguous block. This option allows
+	  overriding the default setting when ability to allocate very
+	  large blocks of physically contiguous memory is required.
+
+	  Don't change if unsure.
+
 config ARCH_FLATMEM_ENABLE
 	def_bool !NUMA
 
-- 
2.47.0

